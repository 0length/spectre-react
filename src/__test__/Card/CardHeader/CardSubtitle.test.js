import React from 'react'
import { shallow } from 'enzyme'

import { CardSubtitle } from '../../..'

describe('CardSubtitle', () => {
  it('Should render default .card-subtitle markup', () => {
    const wrapper = shallow(<CardSubtitle />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('card-subtitle')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <CardSubtitle>
        <p>hello world</p>
      </CardSubtitle>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<CardSubtitle className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('card-subtitle')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<CardSubtitle renderAs='main'>custom</CardSubtitle>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('card-subtitle')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<CardSubtitle />)

    expect(wrapper).toMatchSnapshot()
  })
})
