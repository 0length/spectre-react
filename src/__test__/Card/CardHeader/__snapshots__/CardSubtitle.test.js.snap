// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSubtitle Should match the snapshot 1`] = `
<div
  className="card-subtitle"
/>
`;
